proposals/restructure-KF-docs-proposal.md

```diff
@@ -0,0 +1,109 @@
+This is an initial proposal, to be iterated on, following the discussion in #2293.
```

Comment:
Could you post the actual link here for `#2293` since it's not in this repo?

Reply:
Sure!